examples/example-2/client/example-2.js

```diff
@@ -0,0 +1,95 @@
+// Create the Invoice constructor
+Invoice = ReactiveConstructor(function Invoice ( initData ) {
+  
+  // You need to set this.initData to the passed initData
+  // (Maybe this will be updated in the future.)
+  this.initData = initData;
+  
+  // Define the structure of this object.
+  // You can have multiple types of invoices if you want,
+  // but in this case we'll only have one.
+  this.typeStructure = [{
+    type: 'invoice',
+    fields: {
+      items: [InvoiceListItem],
+    }
+  }];
+  
+  // Here's a method for getting the total value of the
+  // invoice list items price value
+  this.getEndPrice = function() {
+    var items = this.getReactiveValue( 'items' );
+    return _.reduce(items, function( memo, item ){
+      return memo + item.getReactiveValue('price');
+    }, 0);
+  };
+  
+  // This method needs to be called like this.
+  // (Maybe this will be updated in the future.)
+  this.initReactiveValues();
+
+});
+
+// …and setup the InvoiceListItem constructor
+InvoiceListItem = ReactiveConstructor(function InvoiceListItem ( initData ) {
```

Comment:
Missing 'new' prefix when invoking a constructor.<br>'InvoiceListItem' is not defined.<br>'ReactiveConstructor' is not defined.